### ISSUE TITLE
[REVIEW]: An open source crash course on parameter estimation of computational models using a Bayesian optimization approach

### ISSUE THREAD

**Issue by whedon** [openjournals/jose-reviews#89]
**Submitting author:** @mbarzegary (<a href="http://orcid.org/0000-0002-1456-0610">Mojtaba Barzegari</a>)
**Repository:** <a href="https://github.com/mbarzegary/educational-bayesian" target ="_blank">https://github.com/mbarzegary/educational-bayesian</a>
**Version:** v2.0
**Editor:** @magsol
**Reviewers:** @jcapriot, @arokem
**Archive:** <a href="https://doi.org/10.5281/zenodo.5013865" target="_blank">10.5281/zenodo.5013865</a>

## Status

[![status](https://jose.theoj.org/papers/a183b3613fc340277f964d1dd2de7814/status.svg)](https://jose.theoj.org/papers/a183b3613fc340277f964d1dd2de7814)

Status badge code:

```
HTML: <a href="https://jose.theoj.org/papers/a183b3613fc340277f964d1dd2de7814"><img src="https://jose.theoj.org/papers/a183b3613fc340277f964d1dd2de7814/status.svg"></a>
Markdown: [![status](https://jose.theoj.org/papers/a183b3613fc340277f964d1dd2de7814/status.svg)](https://jose.theoj.org/papers/a183b3613fc340277f964d1dd2de7814)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/mbarzegary/educational-bayesian" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@e-dub & @jcapriot, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @magsol know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSE conflict of interest policy](https://github.com/openjournals/jose/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSE code of conduct](https://jose.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source for this learning module available at the <a target="_blank" href="https://github.com/mbarzegary/educational-bayesian">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of a standard license? (OSI-approved for code, Creative Commons for content)
- [ ] **Version:** v2.0
- [ ] **Authorship:** Has the submitting author (@mbarzegary) made visible contributions to the module? Does the full list of authors seem appropriate and complete?

### Documentation

- [ ] **A statement of need:** Do the authors clearly state the need for this module and who the target audience is?
- [x] **Installation instructions:** Is there a clearly stated list of dependencies?
- [ ] **Usage:** Does the documentation explain how someone would adopt the module, and include examples of how to use it?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the module 2) Report issues or problems with the module 3) Seek support

### Pedagogy / Instructional design (Work-in-progress: reviewers, please comment!)

- [x] **Learning objectives:** Does the module make the learning objectives plainly clear? (We don't require explicitly written learning objectives; only that they be evident from content and design.)
- [x] **Content scope and length:** Is the content substantial for learning a given topic? Is the length of the module appropriate?
- [ ] **Pedagogy:** Does the module seem easy to follow? Does it observe guidance on cognitive load? (working memory limits of 7 +/- 2 chunks of information)
- [ ] **Content quality:** Is the writing of good quality, concise, engaging? Are the code components well crafted? Does the module seem complete?
- [ ] **Instructional design:** Is the instructional design deliberate and apparent? For example, exploit worked-example effects; effective multi-media use; low extraneous cognitive load.

### JOSE paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [ ] **A statement of need:** Does the paper clearly state the need for this module and who the target audience is?
- [ ] **Description:** Does the paper describe the learning materials and sequence?
- [ ] Does it describe how it has been used in the classroom or other settings, and how someone might adopt it?
- [ ] Could someone else teach with this module, given the right expertise?
- [ ] Does the paper tell the "story" of how the authors came to develop it, or what their expertise is?
- [ ] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSE conflict of interest policy](https://github.com/openjournals/jose/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSE code of conduct](https://jose.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source for this learning module available at the <a target="_blank" href="https://github.com/mbarzegary/educational-bayesian">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of a standard license? (OSI-approved for code, Creative Commons for content)
- [x] **Version:** v2.0
- [x] **Authorship:** Has the submitting author (@mbarzegary) made visible contributions to the module? Does the full list of authors seem appropriate and complete?

### Documentation

- [x] **A statement of need:** Do the authors clearly state the need for this module and who the target audience is?
- [x] **Installation instructions:** Is there a clearly stated list of dependencies?
- [x] **Usage:** Does the documentation explain how someone would adopt the module, and include examples of how to use it?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the module 2) Report issues or problems with the module 3) Seek support

### Pedagogy / Instructional design (Work-in-progress: reviewers, please comment!)

- [x] **Learning objectives:** Does the module make the learning objectives plainly clear? (We don't require explicitly written learning objectives; only that they be evident from content and design.)
- [x] **Content scope and length:** Is the content substantial for learning a given topic? Is the length of the module appropriate?
- [x] **Pedagogy:** Does the module seem easy to follow? Does it observe guidance on cognitive load? (working memory limits of 7 +/- 2 chunks of information)
- [x] **Content quality:** Is the writing of good quality, concise, engaging? Are the code components well crafted? Does the module seem complete?
- [x] **Instructional design:** Is the instructional design deliberate and apparent? For example, exploit worked-example effects; effective multi-media use; low extraneous cognitive load.

### JOSE paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Does the paper clearly state the need for this module and who the target audience is?
- [x] **Description:** Does the paper describe the learning materials and sequence?
- [x] Does it describe how it has been used in the classroom or other settings, and how someone might adopt it?
- [x] Could someone else teach with this module, given the right expertise?
- [x] Does the paper tell the "story" of how the authors came to develop it, or what their expertise is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @e-dub, @jcapriot it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/jose-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/jose-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1137/1.9780898717921 is OK
- 10.1007/978-94-009-0909-0 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://raw.githubusercontent.com/openjournals/jose-papers/jose.00089/jose.00089/10.21105.jose.00089.pdf)

**Comment by magsol** [comment]
Hey @e-dub and @jcapriot , wanted to check in and see how things were going. Any kind of ETA or obstacles I can help with?

**Comment by magsol** [comment]
Hey @e-dub and @jcapriot, please advise if you're running into any blockers that I can help knock down. Would love to get the reviews for this in soon.

**Comment by jcapriot** [comment]
@magsol Sorry for the lack of response, I will get this today and work through it. It looks like the original accept link has expired for me though.

**Comment by magsol** [comment]
Hi @e-dub and @jcapriot , how are things going? Anything I can help with?

**Comment by magsol** [comment]
Hi @jcapriot:

Any word on when you could finish the review? I'd really like to get this moving.

**Comment by mbarzegary** [comment]
@magsol may I start to answer the comments? or I should wait for the second review to come. it's almost one year since I started this process.

**Comment by magsol** [comment]
@mbarzegary Yes, please respond to the comments! I had been planning to follow up on this thread and see if you even wanted the review to go forward since I hadn't heard anything from you responding to existing feedback. If you want to continue, then yes--please respond to the feedback already given, and I will track down another reviewer if @e-dub does not respond ASAP.

**Comment by e-dub** [comment]
> @mbarzegary Yes, please respond to the comments! I had been planning to follow up on this thread and see if you even wanted the review to go forward since I hadn't heard anything from you responding to existing feedback. If you want to continue, then yes--please respond to the feedback already given, and I will track down another reviewer if @e-dub does not respond ASAP.

Respond??? See [here](https://github.com/mbarzegary/educational-bayesian/issues)

**Comment by magsol** [comment]
@e-dub I'm so sorry! I'd blame it on "dad brain" (I'm on paternity leave at the moment so everything is super fuzzy) but I should have just scrolled up a bit further; my apologies.

@mbarzegary Please check out the "issues" tab for your submission, under which @e-dub has submitted numerous pieces of feedback you can begin to address.

> @magsol Sorry for the lack of response, I will get this today and work through it. It looks like the original accept link has expired for me though.

@jcapriot Here is the link https://github.com/mbarzegary/educational-bayesian/

**Comment by mbarzegary** [comment]
Hi all.
Thank you for the valuable feedback. I addressed the review comments.

**Comment by magsol** [comment]
@e-dub: how does the feedback look?

@jcapriot Any movement on your review?

**Comment by jcapriot** [comment]
I'm moving on it now, again I'm very sorry for my delay in getting through this as my life has been in a constant flux recently.
Again, I will need the link to accept the invitation, as that seems to have expired for me. (Not the link to the repository) As I cannot edit the checklist on this page.

**Comment by magsol** [comment]
@jcapriot No worries; 2020 has been a barnburner for all of us in many different ways... to put it lightly.

Hmm; I'm relatively new as an editor so this is the first time I've seen this problem. From what I can tell, it looks like you need to be 1) logged in (clearly you are), and 2) accept the invite at this link: https://github.com/openjournals/joss-reviews/invitations .

If that still doesn't solve the problem, let me know and I'll start pinging people up the chain of command.

**Comment by jcapriot** [comment]
I keep getting this error when clicking the link.
<img width="1021" alt="Screen Shot 2020-12-11 at 2 39 49 PM" src="https://user-images.githubusercontent.com/16258927/101961556-cb907880-3bbe-11eb-85e6-e3916ad9c65e.png">

**Comment by magsol** [comment]
Hi @openjournals/joss-eics:

We're running into a problem where one of the reviewers can't seem to modify the review checklist in this ticket. I see @jcapriot 's account listed as "Awaiting jcapriot's response" under the "Manage Access" page for `jose-reviews` repository, but [you can see from the above post](https://github.com/openjournals/jose-reviews/issues/89#issuecomment-743464450) that he's not having any success accepting the invitation.

I'm out of ideas. Could you provide some assistance?

**Comment by kthyng** [comment]
Hm I didn't seem @jcapriot for JOSS access. I'll invite again just in case that fixes it. @jcapriot if you have more than one github account, make sure you are using the correct login.

**Comment by kthyng** [comment]
@whedon re-invite @jcapriot as reviewer

**Comment by whedon** [comment]
I'm sorry @kthyng, I'm afraid I can't do that. That's something only editors are allowed to do.

**Comment by kthyng** [comment]
Oh! This is JOSE not JOSS! You need to ping @openjournals/jose-eics instead.

**Comment by magsol** [comment]
@kthyng Oh geez, I'm sorry about that. Autocorrect for the loss :P

@openjournals/jose-eics , [please see this message above](https://github.com/openjournals/jose-reviews/issues/89#issuecomment-745495226) (ignore that I accidentally tagged joss-eics).

**Comment by magsol** [comment]
@jcapriot I was able to cancel and resubmit your invitation to modify the JOSE GitHub repos. Let me know if this works.

**Comment by magsol** [comment]
@e-dub Any suggestions on the feedback provided by @mbarzegary ?

**Comment by magsol** [comment]
Hi all  👋

The reviews are going silent again. **Please respond if you want to continue the review process, otherwise we'll have to end review without acceptance.**

@e-dub: Can you look over the responses to your initial feedback provided by @mbarzegary and verify that they satisfy your initial concerns?

@jcapriot: Please check for an invitation to modify the JOSE GitHub repos again, as I canceled and resubmitted your request 12 days ago. Notify me ASAP if you still cannot perform edits; otherwise, please proceed with your review.

**Comment by mbarzegary** [comment]
a rejection after 10 months is very disappointing for me :(, but apparently this is the dark side of the "community review". 
@magsol may I resubmit the paper with you again as the editor?

**Comment by labarba** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/jose-papers/jose.00089/jose.00089/10.21105.jose.00089.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/jose-papers/blob/jose.00089/jose.00089/10.21105.jose.00089.pdf) :page_facing_up: :point_left:

**Comment by labarba** [comment]
@whedon check repository

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.07 s (83.9 files/s, 18008.0 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
Jupyter Notebook                 1              0            828            233
Markdown                         3             44              0             88
TeX                              1              6              0             60
Python                           1              7              6             16
-------------------------------------------------------------------------------
SUM:                             6             57            834            397
-------------------------------------------------------------------------------


Statistical information for the repository '4431d99643a38fd4b42f4759' was
gathered on 2021/02/21.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
mbarzegary                       1            29              0          100.00

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
mbarzegary                   29          100.0          0.0               20.69
```

**Comment by labarba** [comment]
@mbarzegary — Can you confirm here that you have addressed all the comments of the first reviewer (@e-dub) and are ready for a second look? (I have reached out by email.)

**Comment by labarba** [comment]
Hi @jcapriot — I see that you opened some issues in the target repository back in December. Can you check of items in your review checklist now? Are the permission problems solved? Let us know if you can give this submission another look now.

**Comment by mbarzegary** [comment]
dear @labarba 
yes, I confirm that I have addressed them all. I reopened all the issues to be closed by @e-dub after confirmation.

**Comment by labarba** [comment]
Hi @e-dub 👋  — the author says they have addressed all the issues you opened in the target repository. Could you give this a second look now, and see if you can check off (some of) the remaining items in your checklist?

**Comment by labarba** [comment]
@magsol — I have searched quite a bit, but could not find an email for @jcapriot... do you have one? I found in his LinkedIn profile that he is now a postdoc at UBC, after graduating last year. If he does not respond anymore, we will need a replacement reviewer. Please have a look around...

**Comment by jcapriot** [comment]
@labarba I still don't have the ability to edit my boxes on this page, and I've never actually received the link to be able to join the JOSE repo to edit them. It still says that I do not have permissions, and I have tried many times to gain permissions. I have not seen it come through. I should have my email listed on my GitHub account here. I did add comments and errors in a couple issues on the target repo that look to have been addressed.

**Comment by labarba** [comment]
@whedon re-invite @jcapriot as reviewer

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@jcapriot please accept the invite by clicking this link: https://github.com/openjournals/jose-reviews/invitations

**Comment by labarba** [comment]
hey @jcapriot 👋 Nice to see you pop in here! Try the link ☝️

**Comment by jcapriot** [comment]
Thank you, sorry for all of the confusion :/

**Comment by magsol** [comment]
Thanks @labarba!

@jcapriot Let me know if you need anything else to complete your review.

@e-dub Please follow up regarding whether @mbarzegary has adequately addressed your feedback, or if anything else is needed. I'd like to finalize your review as soon as possible.

**Comment by magsol** [comment]
Hi folks,

We really need to move on this.

@jcapriot: Please let me know when you can complete your review. If it's finished, please let me know so we can finalize your end.

@e-dub Please follow up on whether @mbarzegary has adequately addressed your reviewer feedback. If so, I'd like to finalize your portion of the review.

Let's get this done as soon as possible. Thank you.

**Comment by magsol** [comment]
@mbarzegary We're between a bit of a rock and a hard place. Are you waiting on either of the reviewers at this point?

**Comment by mbarzegary** [comment]
@magsol technically speaking, no. the second reviewer has left some of the checkboxes unchecked without creating issues in the repository, so I don't know his comments on the items to fix them. I have fixed most of the unchecked items according to the comments of reviewer 1.

**Comment by magsol** [comment]
Hi @jcapriot 👋 Please see the note from the author below regarding your review checklist:

> @magsol technically speaking, no. the second reviewer has left some of the checkboxes unchecked without creating issues in the repository, so I don't know his comments on the items to fix them. I have fixed most of the unchecked items according to the comments of reviewer 1.

Please respond as soon as you can with whatever help you may need from me or the author, and whatever additional steps you have yet to take.

**Comment by magsol** [comment]
@jcapriot If you cannot complete your review, please advise ASAP so we can find another reviewer.

**Comment by jcapriot** [comment]
Hello everyone, I have gone through and updated the checkboxes. There is still an unresolved issue related to my review on the target repo, (from Dec 10th: https://github.com/mbarzegary/educational-bayesian/issues/20) that I feel is the only place where this module is significantly lacking.

**Comment by mbarzegary** [comment]
Thanks @jcapriot 
@magsol I addressed the issue and released a new version

**Comment by magsol** [comment]
Excellent, thank you for your quick work @mbarzegary!

@jcapriot please have a look at the changes and report back if they address your concerns.

**Comment by jcapriot** [comment]
The improvements made to the tutorial are very helpful in understanding the workflow, and they also now show a consistent pattern to the logic for choosing them. They have addressed all of my concerns now.

**Comment by magsol** [comment]
Excellent! Thank you @jcapriot for your quick turnaround. 

@e-dub We are ready for your sign-off on publication readiness, or for any further pieces of feedback you may have.

**Comment by labarba** [comment]
Hi @e-dub 👋  — We're hoping to get your final checks on this JOSE submission to proceed to publication. Can you pop in the Review issue and give things a new look? Thank you for your contribution!

**Comment by labarba** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/jose-papers/jose.00089/jose.00089/10.21105.jose.00089.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/jose-papers/blob/jose.00089/jose.00089/10.21105.jose.00089.pdf) :page_facing_up: :point_left:

**Comment by labarba** [comment]
Hello @e-dub : I have seen the issue you opened in the target repository:
https://github.com/mbarzegary/educational-bayesian/issues/21

Given that the comment is a harsh critique of the work, bereft of constructive suggestions, I have taken the time to read the Jupyter notebook from top to bottom (a careful reading took 34 minutes) . Apart from fairly minor typographical and grammatical booboos that could be fixed in one copy editing pass, I found it to be readable and well put together. I am not an expert on optimization or inverse problems, yet I followed the argument and formed a mental idea of how even I could teach with this material. I would assign the introduction to read ahead of lecture, and would prepare myself to go through the worked example in a live-coding format, with students following along. I estimate this would take about 2 hours of a live session, after which students should indeed get a good sense for how these methods work. A more thorough review might lead me to make suggestions or ask clarifications from the author about code improvements (e.g., I might ask why go through lists when handling the data and not go straight to NumPy arrays). Hardly a "complete rewrite" job!

@e-dub, I ask if you are able to enter into a constructive process with this review, or if you find that you don't see a path for that. We are grateful for your efforts and willingness to make a contribution to JOSE. If you'd prefer to bow out, I appreciate you letting me know sooner rather than later.

@mbarzegary — After reading the lesson, my assessment is that it is fairly short, compared with the size of learning module we target in JOSE. That does not make it unpublishable. I would like to suggest augmenting it slightly with peppered  short student exercises, and adding at the end a brief section describing what a student mini-project might look like, for those interested in more lasting retention.

**Comment by e-dub** [comment]
I stand by my criticism and reject your comment of "bereft of constructive suggestions".  Constructive suggestions were made both in this cycle as well in the first, including the laundry list that is found in the specific comment in question.

If you find that this criticism overly harsh (which I indeed do not) and that it is unnecessarily holding up a work that is otherwise publication ready, feel free to replace me as reviewer.

**Comment by labarba** [comment]
Thanks for popping back in here to continue the conversation.

Your post on the issue tracker of the target repo has assessments like:

>I find this document extremely difficult to read and in fact would recommend a full rewrite. The motivation is lacking, both in regards to someone who may teach, in addition to those who would be interested in taking this crash course. 

Suggesting a full rewrite is quite a harsh. You say the motivation is lacking: what would make it better? (Is this referring to the lesson or to the paper?).

>In my eyes, a crash course should be independent in it self, this is not the case. 

How is it not the case? Despite from the technical jargon contained in the las paragraph of the introduction, I found the lesson self-contained. What is missing, in your assessment?

In other words, the author needs your assessments to be followed by requests.

**Comment by mbarzegary** [comment]
dear @labarba,
I addressed the issues you mentioned. I added a couple of exercises and a mini-project. Moreover, I modified the last paragraph of the introduction section to make it less technical.

**Comment by arokem** [comment]
At the request of the EiC and editor for this submission, I have reviewed this article. Here is my assessment.

# Review

## General comments.

This paper introduces materials for a "crash course" in Bayesian optimization. The ideas are clearly important and effective materials for teaching this topic should be helpful. The notebook comprising the materials contains useful examples and walks the learner through these examples step by step, in a helpful manner. The inclusion of exercises is beneficial. After some small debugging (see below), the code runs through with no problem. In general, I find some parts harder to read through because terminology is not always defined and sometimes used inconsistently (e.g., "variables"/"parameters").

## Comments on the notebook:

First a technical comment: Code cell 6 contains an un-needed import of the `hyperopt.random` sub-module. This raised an error on my machine because of changes in more recent versions of hyperopt. So, removing this import will make it possible to run this. The same cell also imports `hyperopt.Trials`, which is also unused. So, this line of code could be simplified to `from hyperopt import hp, tpe, fmin`, without affecting subsequent code.

The first section repeats itself a bit, and it takes a little while until the author tells us what the tutorial will be about. In the end of the section, you finally learn that the goal of the tutorial is specifically to teach Bayesian optimization (as the title also suggests). Though they appear in the paper, I think that it would be great to write a few learning objectives at the top of the notebook, that will help learners and instructors calibrate their expectations. You can find some information about writing effective learning objectives [here](https://ii.library.jhu.edu/2016/07/20/writing-effective-learning-objectives/). The first section is also a bit confusing because of the density of new terms that are introduced, some without an explicit definition (e.g., "calibration", "variables", etc.; see general comments above).

It would be good to change function/variable names to use PEP8 (i.e., snake_case, rather than camelCase). For example: "prepare_data_frame" rather than "prepareDataFrame". This will improve readability and teach learners best practices in software, while also teaching them optimization.

### Minor comments:

- In the first section, the phrase: "best available deals" doesn't make sense.

- There is an extraneous double quote after the sentence: "...especially when we are working with heavy computational models!" in section 2 of the notebook ("Bayesian optimization").

- Section 2: "If you interested" should be "If you are interested".

- The `%matplotlib inline` cell magic is not needed (this has been the default behavior of the notebook for quite a while).

- "The stored data can be plotted using Matplotlib library." => "The stored data can be plotted using the Matplotlib library."

- "I said this to assert that this problem is complex enough to be considered like a case with four unknown parameters." is an awkward turn of phrase. Consider changing to something like: "This demonstrates that the problem we will analyze here is complex enough -- at least as complex as other cases with four unkown parameters" or something like that.

- Code cell 7 erroneously defines the inputs of the rmse functions as lists. This
function would error out if provided with lists as inputs, because `-` is not defined as a binary operator for pairs of lists.

- Code cell 8 uses the built-in function name `input` as a variable name, which should be avoided.

- In the text after cell 10, "`runOptimization` first reset the global `iterNumber` variable" should be "`runOptimization` first resets the global `iterNumber` variable".

- In the text after code cell 13, "real-world axes titles" should be "real-world axes labels".

- In the "Exercises" section, "more extensive objective function" should probably be "more expensive objective function".

- In the "Mini project" section contains the sentence "The, you should be able to run the code and optimize our fitting test-case using Python". I think that first word is not needed?

- Same section, the sentence "If you think this first step is a bit difficult, you have a look at this repository, in which the same code is implemented ready-to-use for any problem." Should be "If you think this first step is a bit difficult, you can have a look at this repository, in which the same code is implemented ready-to-use for any problem." 

- Same section: "whether you can obtain the most optimal value of the parameters or not". What does it mean when you can't find optimal parameters?

## Comments on the manuscript:

It would be good to define the prerequisites for the course, in terms of what learners need to know before they learn this material. For example, is knowledge of PDEs required. One might get that impression from reading the example (maybe consider simplifying this example?), but presumably the answer is no.

### Minor comments:

The name of the conference for the Bergstra et al. reference contains the phrase "International Conference on" twice.

Chen et al. is missing a DOI (I believe this should be https://doi.org/10.1037/edu0000018).

**Comment by mbarzegary** [comment]
dear @arokem,
Thank you for your constructive comments. I addressed them through the following distinct commits:

* [change variable/function names to PEP8](https://github.com/mbarzegary/educational-bayesian/commit/9d73bc828219744c953f829d377389e1f85834d9
)
* [fix python code issues](https://github.com/mbarzegary/educational-bayesian/commit/f8888c45b8637f92684c05a5d0e6c02470b1e6fb
)
* [correct ambiguous sentences](https://github.com/mbarzegary/educational-bayesian/commit/330de6838bb4b1436bb37fe242d487eb2bd87872)
* [improve introduction section and add learning objective](https://github.com/mbarzegary/educational-bayesian/commit/31947bf76105c1d6e381a0aa296392a3fcb60435)
* [add prerequisites and update objectives](https://github.com/mbarzegary/educational-bayesian/commit/146137b89339bf961f26ce86e14c11bdca71b78c)
* [fix references issues and add missing dois](https://github.com/mbarzegary/educational-bayesian/commit/212acd73b50595dc99eaa1be60d25e03e6a86675)

Moreover, I released a new version.

**Comment by magsol** [comment]
Thanks for your quick integration work @mbarzegary.

@arokem Do these changes address your critiques?

**Comment by arokem** [comment]
Yeah. Looks great! This is ready for publication, as far as I can tell.

**Comment by labarba** [comment]
@whedon add @arokem as reviewer

**Comment by whedon** [comment]
OK, @arokem is now a reviewer

**Comment by labarba** [comment]
@arokem — I'm adding you as reviewer so that you will be credited for your contribution on the paper's landing page. Thank you!

**Comment by magsol** [comment]
Thank you @labarba , and thank you @arokem for your review! 

Let's see if we can get the acceptance process rolling.

**Comment by magsol** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/jose-papers/jose.00089/jose.00089/10.21105.jose.00089.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/jose-papers/blob/jose.00089/jose.00089/10.21105.jose.00089.pdf) :page_facing_up: :point_left:

**Comment by magsol** [comment]
Ok, all the links look good to me.

@whedon check references

**Comment by magsol** [comment]
I don't see any misspellings or bad citations, but it does seem like one of the citations lacks a DOI: Grindrod 1996. I can't seem to find the article online, so that may be why, but still worth double-checking.

@mbarzegary Please do the following:
- [ ] make a final pass for any typos I may have missed
- [ ] see if the Grindrod 1996 has a DOI, but if not, no worries (just want to make sure my Google skills aren't failing me)
- [ ] create a tagged release and archive
- [ ] report the version number and archive DOI here

Once you've finished these above items we can proceed with acceptance and publication.

**Comment by mbarzegary** [comment]
Thanks, @magsol 
You are right, the citation doesn't have a DOI, and that's why I didn't provide it.

The version number is **v2.0**: [https://github.com/mbarzegary/educational-bayesian/releases/tag/v2.0](https://github.com/mbarzegary/educational-bayesian/releases/tag/v2.0)
The archive DOI is **10.5281/zenodo.5013865**: [https://doi.org/10.5281/zenodo.5013865
](https://doi.org/10.5281/zenodo.5013865)

**Comment by magsol** [comment]
Thanks @mbarzegary!

**Comment by magsol** [comment]
@whedon set 10.5281/zenodo.5013865 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.5013865" target="_blank">10.5281/zenodo.5013865</a> is the archive.

**Comment by magsol** [comment]
@whedon set v2.0 as version

**Comment by whedon** [comment]
OK. v2.0 is the version.

**Comment by magsol** [comment]
@whedon accept

**Comment by whedon** [comment]
To recommend a paper to be accepted use `@whedon recommend-accept`

**Comment by magsol** [comment]
@whedon recommend-accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1137/1.9780898717921 is OK
- 10.1007/978-94-009-0909-0 is OK
- 10.1016/B978-0-12-411557-6.00005-7 is OK
- 10.1016/S0307-904X(01)00010-5 is OK
- 10.1037/edu0000018 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/jose-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/jose-papers/pull/57

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/jose-papers/pull/57, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by labarba** [comment]
@mbarzegary : Please merge my PR with a handful of copy edits, and we'll publish this! *Thank you for your patience.*

**Comment by mbarzegary** [comment]
@labarba thanks. I merged the edits.

**Comment by magsol** [comment]
@whedon remove @e-dub as reviewer

**Comment by whedon** [comment]
OK, @e-dub is no longer a reviewer

**Comment by labarba** [comment]
@whedon accept

**Comment by whedon** [comment]
To recommend a paper to be accepted use `@whedon recommend-accept`

**Comment by labarba** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/jose-papers/jose.00089/jose.00089/10.21105.jose.00089.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/jose-papers/blob/jose.00089/jose.00089/10.21105.jose.00089.pdf) :page_facing_up: :point_left:

**Comment by labarba** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSE!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/jose-papers/pull/58
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/jose.00089](https://doi.org/10.21105/jose.00089)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by labarba** [comment]
Congratulations,  @mbarzegary, your JOSE paper is published 🚀 
(thank you for your patience as we navigated reduced capacity of volunteers to contribute)

Huge thanks to the Editor: @magsol, and thank you to the Reviewers: @jcapriot, @arokem 🙏

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://jose.theoj.org/papers/10.21105/jose.00089/status.svg)](https://doi.org/10.21105/jose.00089)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/jose.00089">
  <img src="https://jose.theoj.org/papers/10.21105/jose.00089/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://jose.theoj.org/papers/10.21105/jose.00089/status.svg
   :target: https://doi.org/10.21105/jose.00089
```
This is how it will look in your documentation:

[![DOI](https://jose.theoj.org/papers/10.21105/jose.00089/status.svg)](https://doi.org/10.21105/jose.00089)

**We need your help!**

Journal of Open Source Education is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. 
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-jose

**Comment by labarba** [comment]
:loudspeaker:  New Reviewer Sign-up Form!
https://forms.gle/Rizd3TcHnQKhrbYY7
